src/components/HomePage.jsx

```diff
@@ -1,19 +1,225 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Navbar from '../views/Navbar.jsx';
+import Footer from '../views/Footer.jsx';
+import Category from '../views/Category.jsx';
+import { getArticles, getArticlesCategory } from '../actions';
+import FooterImage from '../../public/images/background/Image.png';
+import Articles from '../views/Articles.jsx';
+import Authors from '../views/Authors.jsx';
+import PlaceholderLoader from '../views/placeholderLoader.jsx';
+import { getAverageRatings, totalRatings } from '../utils/getAverageRatings';
 
-class HomePage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
+/**
+ * @description - returns the home page view
+ * @param {props} articles - all authors article
+ * @param {props} articlesCategory - articles categories
+ * @return {JSX} 
+ */
+export class HomePage extends Component {
+  state = {
+    fiveStarAuthors: [],
   }
+  async componentDidMount() {
+    const { getArticles, getArticlesCategory } = this.props;
+    await Promise.all([getArticles(), getArticlesCategory()]);
+  }
+
+  /**
+   * @description Gets the averahe ratings of author's article
+   * @param {object} article - each authors article
+   * @return {array} fiveStaAauhtors - author's rated highly
+   */
+  averageRatings = (article) => {
+    getAverageRatings(article);
+    totalRatings.map(async rating => {
+      if (rating.averageRatings === 5) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 4) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 3) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 2) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      } else if (rating.averageRatings === 1) {
+        if (!this.found(rating)) {
+          await this.setAuthorsState(rating);
+        }
+      }
+    });
+  }
+
+  /**
+ * @description set the state with start authors
+ * @return {void}
+ */
+  setAuthorsState = (rating) => (this.setState({
+    fiveStarAuthors: [...this.state.fiveStarAuthors, rating],
```

Comment:
#### Comment
- Can you explain why you are using the spread operator here?

Reply:
Yeah, I'm using it to update my `fiveStarAuthors` state